packages/medusa/src/services/shipping-profile.ts

```diff
@@ -0,0 +1,513 @@
+        this.shippingProfileRepository_
+      )
+
+      // Should not fail, if profile does not exist, since delete is idempotent
+      const profile = await profileRepo.findOne({ where: { id: profileId } })
+
+      if (!profile) {
+        return Promise.resolve()
+      }
+
+      await profileRepo.softRemove(profile)
+
+      return Promise.resolve()
+    })
+  }
+
+  /**
+   * Adds a product to a profile. The method is idempotent, so multiple calls
+   * with the same product variant will have the same result.
+   * @param profileId - the profile to add the product to.
+   * @param productId - the product to add.
+   * @return the result of update
+   */
+  async addProduct(
+    profileId: string,
+    productId: string
+  ): Promise<ShippingProfile> {
+    return await this.atomicPhase_(async (manager) => {
+      await this.productService_
+        .withTransaction(manager)
+        .update(productId, { profile_id: profileId })
+
+      return await this.retrieve(profileId)
+    })
+  }
+
+  /**
+   * Adds a shipping option to the profile. The shipping option can be used to
+   * fulfill the products in the products field.
+   * @param profileId - the profile to apply the shipping option to
+   * @param optionId - the option to add to the profile
+   * @return the result of the model update operation
+   */
+  async addShippingOption(
+    profileId: string,
+    optionId: string
+  ): Promise<ShippingProfile> {
+    return await this.atomicPhase_(async (manager) => {
+      await this.shippingOptionService_
+        .withTransaction(manager)
+        .update(optionId, { profile_id: profileId })
+
+      const updated = await this.retrieve(profileId)
+      return updated
+    })
+  }
+
+  /**
+   * Decorates a profile.
+   * @param profile - the profile to decorate.
+   * @param fields - the fields to include.
+   * @param expandFields - fields to expand.
+   * @return return the decorated profile.
+   */
+  async decorate(
+    profile: ShippingProfile,
+    fields: string[],
+    expandFields: string[] = []
+  ): Promise<ShippingProfile> {
+    const requiredFields: string[] = ["_id", "metadata"]
+    const decorated = _.pick(profile, fields.concat(requiredFields))
+
+    if (expandFields.includes("products") && profile.products) {
+      decorated.products = await Promise.all(
+        profile.products.map((pId) => this.productService_.retrieve(pId.id))
+      )
+    }
+
+    if (expandFields.includes("shipping_options") && profile.shipping_options) {
+      decorated.shipping_options = await Promise.all(
+        profile.shipping_options.map((oId) =>
+          this.shippingOptionService_.retrieve(oId)
+        )
+      )
+    }
+    return decorated as ShippingProfile
+  }
+
+  /**
+   * Returns a list of all the productIds in the cart.
+   * @param cart - the cart to extract products from
+   * @return a list of product ids
+   */
+  getProfilesInCart_(cart: Cart): string[] {
```

Comment:
suggestion: should we also refactor method names per convetion

```suggestion
  protected getProfilesInCart(cart: Cart): string[] {
```

Reply:
Good catch 😄